test/integration/node/lifecycle_test.go

```diff
@@ -44,6 +48,144 @@ const poll = 2 * time.Second
 
 type podCondition func(pod *v1.Pod) (bool, error)
 
+// TestNoExecuteTaintBasedEviction tests taint-based eviction for a node tainted NoExecute
+func TestNoExecuteTaintBasedEviction(t *testing.T) {
```

Comment:
Could this have been added as a test case for `TestTaintBasedEvictions`?

Reply:
We could, but it would risk making the test flaky due to timeout. The current TestTaintBasedEvictions takes already on my machine 41s which is above the default timeout of 30s.

Also, the test TestTaintBasedEvictions is ticketed for simplifications here: [Simplify the "TestTaintBasedEvictions" integration test in lifecycle_test.go](https://github.com/kubernetes/kubernetes/issues/111183). I think the new test I introduce could serve as an example of how to refactor the existing one. Once the test is refactored it could take significantly less time (due to elimination of kube scheduler and periodic heartbeats) and then merging the tests could be feasible.

Reply:
When makes the new test also sufficiently different from the existing one is that the test simulates adding a custom NoExecute taint by the user, whereas the existing test awaits (https://github.com/kubernetes/kubernetes/blob/7156c96e5d8f4fe58a36000ff53af7e21743f49b/test/integration/node/lifecycle_test.go#L273) for the taints to be added by the node_lifecycle_controller (https://github.com/kubernetes/kubernetes/blob/7156c96e5d8f4fe58a36000ff53af7e21743f49b/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L708)